src/cephadm/tests/build/test_cephadm_build.py

```diff
@@ -124,3 +134,59 @@ def test_cephadm_build(env, source_dir, tmp_path):
     assert any(e.startswith('markupsafe') for e in zre)
     assert any(e.startswith('cephadmlib') for e in zre)
     assert any(e.startswith('_cephadmmeta') for e in zre)
+
+
+@pytest.mark.parametrize(
+    'env',
+    [
+        'centos-8-plusdeps',
+        'centos-9-plusdeps',
+        'centos-9',
+    ],
+)
+def test_cephadm_build_from_rpms(env, source_dir, tmp_path):
+    res = build_in(
+        env,
+        source_dir,
+        tmp_path,
+        ['-Brpm', '-SCEPH_GIT_VER=0', '-SCEPH_GIT_NICE_VER=foobar'],
+    )
+    if 'plusdeps' not in env:
+        assert res.returncode != 0
+        return
+    binary = tmp_path / 'cephadm'
+    if 'centos-8' in env and sys.version_info[:2] >= (3, 10):
+        # The version of markupsafe in centos 8 is incompatible with
+        # python>=3.10 due to changes in the stdlib therefore we can't execute
+        # the cephadm binary, so we quit the test early.
+        return
```

Comment:
given these tests are using a container to try and build in, are we actually expecting inconsistencies in the python version we find?

Reply:
I could change the test to execute the binary in the container too, but right now it's just executing the binary on the "host" python. I think that maybe it could be done in a follow up pr.